packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -353,15 +629,235 @@ fn expand_input_param(
     }
 }
 
+// @todo This is an experimental support for the new JSON ABI file format.
+// Once this is stable:
+// 1. Delete old one;
+// 2. Rename it to its original name;
+// 3. Write documentation.
+fn _new_expand_input_param(
+    fun: &ABIFunction,
+    type_application: &TypeApplication,
+    kind: &ParamType,
+    types: &HashMap<usize, TypeDeclaration>,
+) -> Result<TokenStream, Error> {
+    match kind {
+        ParamType::Array(ty, _) => {
+            let ty = _new_expand_input_param(fun, type_application, ty, types)?;
+            Ok(quote! {
+                ::std::vec::Vec<#ty>
+            })
+        }
+        ParamType::Enum(_) => {
+            let t = types.get(&type_application.type_field).unwrap(); // temp unwrap
+
+            let ident = ident(&_new_extract_custom_type_name_from_abi_property(
+                t,
+                Some(CustomType::Enum),
+                types,
+            )?);
+            Ok(quote! { #ident })
+        }
+        ParamType::Struct(_) => {
+            let t = types.get(&type_application.type_field).unwrap(); // temp unwrap
```

Comment:
```suggestion
            let t = types.get(&type_application.type_field).unwrap();
```

Reply:
Ah, nice catch, forgot to add my old `@todo` and grep for it before pushing 😂. My intended goal was to actually turn it into a proper `.expect()` with a nice message -- will update it (and the other instances) soon!